src/gpu/intel/compute/device_info.hpp

```diff
@@ -38,6 +38,12 @@ namespace compute {
 
 enum class gpu_arch_t { unknown, xe_lp, xe_hp, xe_hpg, xe_hpc, xe2, xe3 };
 
+// Memory for storing ngen::Product to avoid directly including nGEN because of
+// header dependencies outside of src/gpu/intel.
+struct alignas(int) gpu_product_t {
+    unsigned char data[12];
+};
```

Comment:
A comment for the future: I feel like we can add more isolation to `compute::device_info_t` to ensure its definition is not used outside of `gpu::intel` code. Then we can just use nGEN types here directly.

But IMO, this is not worth it until we touch some related code.